src/interactive-cli.ts

```diff
@@ -21,361 +21,374 @@ import { BN } from '@flarenetwork/flarejs/dist'
-      if (isUnclaimed) await claimRewardsPrivateKey(walletProperties.wallet, context)
-    }
-  }
+        }
 
-  // Withdraw funds
-  else if (Object.keys(taskConstants)[10] == task.toString()) {
-    if (walletProperties.wallet == Object.keys(walletConstants)[0] && fileExists("ctx.json")) {
-      const { derivationPath: ctxDerivationPath } = readInfoFromCtx("ctx.json")
-      const amount = await prompts.amount()
-      const withdrawAddress = await prompts.withdrawAddress()
-      const argsWithdraw = [...baseargv.slice(0, 2), taskConstants[task], '-a', `${amount.amount}`, "-t", `${withdrawAddress.address}`,]
-      await program.parseAsync(argsWithdraw)
+        else if (walletProperties.wallet == Object.keys(walletConstants)[1] && fileExists("ctx.json")) {
+          const isContinue = await prompts.forDefiContinue()
+          const txnId = await prompts.transactionId()
+          if (!isContinue.isContinue) {
+            const amount = await prompts.amount()
+            const withdrawAddress = await prompts.withdrawAddress()
+            const argsWithdraw = [...baseargv.slice(0, 2), taskConstants[task], '-a', `${amount.amount}`, "-t", `${withdrawAddress.address}`, "-i", `${txnId.id}`]
+            await program.parseAsync(argsWithdraw)
+            const argsSign = [...makeForDefiArguments("sign", baseargv, txnId.id), "--withdrawal"]
+            await program.parseAsync(argsSign)
+          }
+          else {
+            const argsFetch = [...makeForDefiArguments("fetch", baseargv, txnId.id), "--withdrawal"]
+            await program.parseAsync(argsFetch)
+            const argsSend = [...baseargv.slice(0, 2), taskConstants[task], "-i", `${txnId.id}`, "--send-signed-tx"]
+            await program.parseAsync(argsSend)
+          }
+        }
 
-      const argsSign = [...baseargv.slice(0, 2), "sign-hash", "--derivation-path", ctxDerivationPath!]
-      await program.parseAsync(argsSign)
+        else if (walletProperties.wallet == Object.keys(walletConstants)[2] && walletProperties.network && walletProperties.path) {
+          const context: Context = contextEnv(walletProperties.path, walletProperties.network)
+          const amount = await prompts.amount()
+          const withdrawAddress = await prompts.withdrawAddress()
+          const argsWithdraw = [...baseargv.slice(0, 2), taskConstants[task], '-a', `${amount.amount}`, "-t", `${withdrawAddress.address}`,
+          `--env-path=${walletProperties.path}`, `--network=${walletProperties.network}`, "--get-hacked"]
+          await program.parseAsync(argsWithdraw)
 
-      const argsSend = [...baseargv.slice(0, 2), taskConstants[task],  "--send-signed-tx"]
-      await program.parseAsync(argsSend)
+          const argsSignSend = [...baseargv.slice(0, 2), "signAndSubmit", `--env-path=${walletProperties.path}`, `--network=${walletProperties.network}`, "--get-hacked"]
+          await program.parseAsync(argsSignSend)
 
-    }
+        }
+      }
 
```

Comment:
Can you put comment what this else if condition do? I will ease the work of find the functions